server/internal/cache/middleware.go

```diff
@@ -6,15 +6,19 @@ import (
 
 type RequestOptions func(r *http.Request) *Options
 
+// Middleware propagates a Cache to each request, and
+// injects the caching options for the request.
 func Middleware(c *Cache, opts RequestOptions) func(http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			ctx := ContextWithCache(r.Context(), c)
+
 			if opts != nil {
-				if cOpts := opts(r); opts != nil {
+				if cOpts := opts(r); cOpts != nil {
```

Comment:
This is also not relevant, but is a noticed bugfix where the options should be not nil before applying. (prme-3)